x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java

```diff
@@ -133,6 +149,8 @@ public void parseXContent(XContentParser parser) throws IOException {
                     this.stackTraceIds = parser.text();
                 } else if (REQUESTED_DURATION_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
                     this.requestedDuration = parser.doubleValue();
+                } else if (CUSTOM_COST_FACTOR_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    this.customCostFactor = parser.doubleValue();
```

Comment:
Does it make sense to add this the validation and ensure it is a non-negative value if specified?